test/framework/deployments/externalservice/deployment.go

```diff
@@ -0,0 +1,50 @@
+package externalservice
+
+import (
+	"github.com/onsi/gomega"
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/test/framework"
+)
+
+type ExternalService interface {
+	Init(cluster framework.Cluster, name string, args []string) error
```

Comment:
`ExternalService` should be an interface for the end user that is building a test. Therefore it should only be
```
type ExternalService interface {
  GetExternalAppAddress() string
}
```

instead of init, please pass those arguments (name and args) to `k8SDeployment` and `universalDeployment` objects.

Instead of `Cleanup` there is already `Delete` method here
```
type Deployment interface {
	Name() string
	Deploy(cluster Cluster) error
	Delete(cluster Cluster) error
}
```
(that you implement!). There should be no need to call this cleanup manually. We delete all deployments when we execute `cluster.DismissCluster`